src/sentry/static/sentry/app/views/organizationDiscover/result/utils.jsx

```diff
@@ -16,17 +16,38 @@ const CHART_KEY = '__CHART_KEY__';
  *
  * @param {Array} data Data returned from Snuba
  * @param {Object} query Query state corresponding to data
+ * @param {Object} [options] Options object
+ * @param {Boolean} [options.includePercentages] (default: false) Include aggregation's percentages
  * @returns {Array}
  */
-export function getChartData(data, query) {
+export function getChartData(data, query, options = {}) {
   const {fields} = query;
 
+  const totalsBySeries = new Map();
+
+  if (options.includePercentages) {
+    query.aggregations.forEach(aggregation => {
+      totalsBySeries.set(
+        aggregation[2],
+        data.reduce((acc, res) => {
+          acc += res[aggregation[2]];
+          return acc;
+        }, 0)
+      );
+    });
+  }
+
   return query.aggregations.map(aggregation => {
+    const total = options.includePercentages && totalsBySeries.get(aggregation[2]);
     return {
       seriesName: aggregation[2],
       data: data.map(res => {
         return {
           value: res[aggregation[2]],
+          percentage:
+            options.includePercentages && total
+              ? Math.round(res[aggregation[2]] / total * 10000) / 100
```

Comment:
Is this the format required by echarts? If not I'd suggest doing all the formatting here and formatting as a string e.g. "10.22%"

Reply:
Why format here? Wouldn't it be more flexible for this to remain a number so that components downstream can use the data however they want (e.g. compare/diff 2 sets of percentages)?